src/App.js

```diff
@@ -1,23 +1,38 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
-import { addLocaleData } from 'react-intl';
+import { FormattedMessage, addLocaleData } from 'react-intl';
 import en from 'react-intl/locale-data/en';
 import es from 'react-intl/locale-data/es';
+import { IntlProvider } from 'react-intl';
+import messages_es from './i18n/es.json';
+import messages_en from './i18n/en.json';
+import { flattenMessages } from './utils';
 import jwt_decode from 'jwt-decode';
 import axios from 'axios';
 
 import Header from './components/Header/Header';
 import Footer from './components/Footer/Footer';
 
+const messages = {
+  es: messages_es,
+  en: messages_en,
+};
+
 addLocaleData([...en, ...es]);
 
 class App extends Component {
   constructor() {
     super();
+    //TODO: this hardcoded data will depend by the app language
+    const locale = navigator.language.toLowerCase().split(/[_-]+/)[0] || 'en';
     this.state = {
       user: null,
       loginSession: {},
+      i18n: {
```

Comment:
need this in state?

Reply:
What about using context for language localization? https://reactjs.org/docs/context.html

Reply:
@fcoronelmakingsense I think if the user wants to change the language it should be able to do so, app state seems to be a logical choice for this scenario.

Reply:
yes maybe you are right! 💯

Reply:
Mmm, I heard that it is not a good idea using context API directly, but maybe that info is old.

In this case, we need the state only here, so I think that we do not need _context_ and neither other option.

We could make _messages_ a global variable and simply keep _locale_ in the state. It seemed to me to be more clear and isolated.

Alternative: 

```js
import { flattenMessages } from './utils';
import messages_es from './i18n/es.json';
import messages_en from './i18n/en.json';

// . . .

const messages = {
  es: messages_es,
  en: messages_en,
};

// . . .

class App extends Component {
  constructor() {
    // . . .
    this.state = {
      user: null,
      loginSession: {},
      locale: locale, // 'es' / 'en'
    }
  }

    render() {
      // . . .
      const locale = this.state.locale;
      return (
        <IntlProvider locale={locale} messages={flattenMessages(messages[locale])}>
```

What do you think?

Reply:
But messages will have both languages messages in context right? So we can change, because some time in the near future we will need this feature.

Reply:
@fcoronelmakingsense @cbernat @mvillaseco do you have any opinion?

Reply:
@lucianaDilizio

Reply:
> But messages will have both languages messages in context right? So we can change, because sometime in the near future we will need this feature.

En el ejemplo que puse, _messages_ es una variable global que se usa en la función de _render_. En el código del PR, en la función de _render_ solo se utilizan datos del _state_ (que fueron obtenidos de la variable global)

¿Tienen opinión?

Reply:
Capaz va mejor por fuera del render, no pondría en el state algo que no sea dinámico.

Reply:
Hablamos con @cbernat y coincidimos que la forma en que lo estamos haciendo en el PR parece más limpia.

Respecto al _Context_, no necesitamos usarlo explicitamente, si no me equivoco la librería lo usa internamente, en cualquier caso funciona sin problemas.